s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/S3SourceTask.java

```diff
@@ -107,20 +110,23 @@ public void start(final Map<String, String> props) {
         initializeConverters();
         initializeS3Client();
         this.s3Bucket = s3SourceConfig.getString(AWS_S3_BUCKET_NAME_CONFIG);
-        this.outputWriter = OutputWriterFactory.getWriter(s3SourceConfig);
+        this.transformer = TransformerFactory.getTransformer(s3SourceConfig);
         offsetManager = new OffsetManager(context, s3SourceConfig);
+        fileReader = new FileReader(s3SourceConfig, this.s3Bucket, failedObjectKeys);
         prepareReaderFromOffsetStorageReader();
         this.taskInitialized = true;
     }
 
     private void initializeConverters() {
         try {
             keyConverter = Optional
-                    .of((Converter) s3SourceConfig.getClass("key.converter").getDeclaredConstructor().newInstance());
-            valueConverter = (Converter) s3SourceConfig.getClass("value.converter")
+                    .of((Converter) Class.forName((String) s3SourceConfig.originals().get("key.converter"))
+                            .getDeclaredConstructor()
+                            .newInstance());
```

Comment:
Wrote this initially but then thought I missed an Optional.empty() assignment but now I see that I did not, so adding the comment back:

> Do we expect constructor to return null? Or why do we need optional?

Reply:
This converters instantiation has to be revisited. We already have a ticket https://aiven.atlassian.net/browse/KCON-25